tests/load/pipeline/test_pipelines.py

```diff
@@ -997,3 +997,126 @@ def events_timezone_unset():
 
             values = [r[0].strftime("%Y-%m-%dT%H:%M:%S.%f") for r in rows]
             assert values == output_map[destination]["tables"][t]["timestamp_values"]  # type: ignore
+
+
+@pytest.mark.parametrize(
+    "destination_config",
+    destinations_configs(
+        default_staging_configs=True, default_sql_configs=True, subset=["duckdb", "postgres"]
+    ),
+    ids=lambda x: x.name,
+)
+def test_pipeline_resource_incremental_int_lag(
+    destination_config: DestinationTestConfiguration,
+) -> None:
+    """
+    Test incremental lag behavior when updating previously loaded data.
+
+    Three resources (`r1`, `r2`, `r3`) share the primary key 'id'.
+    - `r1` loads `id=0` and `id=1`.
+    - `r2` adds `id=2` and `id=3`.
+    - `r3`, with lag=1, skips `id=1` and updates entries with `id=2`, `id=3`,
+      and adds a new entry `id=4`. The merge operation reflects these changes.
+
+    We validate that the final dataset correctly orders the updated and new entries,
+    excluding `id=1` from the updates due to the lag.
+    """
+
+    dataset_name = f"{destination_config.destination_name}{uniq_id()}"
+    pipeline = destination_config.setup_pipeline("pipeline", dataset_name=dataset_name)
+
+    name = "items"
+
+    @dlt.resource(name=name, primary_key="id")
+    def r1(_=dlt.sources.incremental("id")):
```

Comment:
I have similar comments as to the test below:
1. use just one resource with "append" write disposition. you can return the same response each time
2. call the resource several time with different lags (use apply_hints to replace incremental)
3. test if expected elements got duplicated
4. IMO you'll have a deduplication bug that I describe above

Reply:
Implemented your comments except for number 2:

- call the resource several time with different lags (use apply_hints to replace incremental)

I have used `@pytest.mark.parametrize("lag", [-1, 10])` to execute the code with different lags instead of using apply_hints.

Also facing the deduplication bug - more info above in the other comments.